packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -40,7 +89,107 @@ export class EmptyModel implements IModel {
  * }
  */
 export class ErrorModel implements IModel {
-    public readonly modelId = 'Error';
+  public readonly modelId = 'Error';
+}
+
+/**
+ * Interface share for creating or importing a model.
+ */
+export interface IModelConstruct extends cdk.IConstruct, IModel {
+
 }
 
-// TODO: Implement Model, enabling management of custom models.
\ No newline at end of file
+/**
+ * Model CFN resource
+ * {
+ *   "Type" : "AWS::ApiGateway::Model",
+ *   "Properties" : {
+ *     "ContentType" : String, <-- Required
+ *     "Description" : String,
+ *     "Name" : String,
+ *     "RestApiId" : String, <-- Required
+ *     "Schema" : JSON object <-- Required
+ *   }
+ * }
+ */
+export class Model extends cdk.Construct implements IModelConstruct {
+
+  /**
+   * Returns a reference to the default Empty model.
+   */
+  public static readonly Empty: IModel = new EmptyModel();
+
+  /**
+   * Returns a reference to the default Error model.
+   */
+  public static readonly Error: IModel = new ErrorModel();
+
+  /**
+   * Import an existing model from another stack
+   * @param scope
+   * @param id
+   * @param props
+   */
+  public static import(scope: cdk.Construct, id: string, props: ModelImportProps): IModelConstruct {
+    return new ImportedModel(scope, id, props);
+  }
+
+  /**
+   * The identifier for this model.
+   */
+  public readonly modelId: string;
+
+  /**
+   * The REST API that this model belongs to.
+   */
+  public readonly restApi: IRestApi;
+
+  constructor(scope: cdk.Construct, id: string, props: ModelProps) {
+    super(scope, id);
+
+    const model = new CfnModel(this, 'Model', {
+      restApiId: props.restApi.restApiId,
+      contentType: props.contentType,
+      schema: props.schema,
+      name: props.name,
+      description: props.description,
+    });
+
+    this.modelId = model.ref;
+    this.restApi = props.restApi;
+  }
+
+  /**
+   * Exports a Model resource from this stack.
+   * @returns Model props that can be imported to another stack.
+   */
+  public export(): ModelImportProps {
+    return {
+      modelId: new cdk.CfnOutput(this, 'ModelId', { value: this.modelId }).makeImportValue().toString(),
+    };
+  }
+
+  /**
+   * Generates the formatted reference to this model, so that another model can reference it, in its schema
+   */
+  public get referenceForSchema(): string {
+    return `https://apigateway.amazonaws.com/restapis/${this.restApi.restApiId}/models/${this.modelId}`;
```

Comment:
Might need to use `cdk.Aws.urlSuffix` instead of `amazonaws.com` here.

Reply:
Sure. I can change that.